models/DataObject/Traits/DefaultValueTrait.php

```diff
@@ -112,21 +113,17 @@ protected function handleDefaultValue(mixed $data, Concrete $object = null, arra
             if (!$this->isEmpty($configuredDefaultValue)) {
                 if ($class && $class->getAllowInherit()) {
                     $params = [];
-
-                    $inheritanceEnabled = Concrete::getGetInheritedValues();
-
                     try {
                         // make sure we get the inherited value of the parent
-                        Concrete::setGetInheritedValues(true);
-
-                        $data = $owner->getValueFromParent($this->getName(), $params);
-                        if (!$this->isEmpty($data)) {
-                            return null;
-                        }
+                        $data = DataObject\Service::useInheritedValues(true, function() use ($owner, $params) {
+                            $data = $owner->getValueFromParent($this->getName(), $params);
+                            if (!$this->isEmpty($data)) {
+                                return null;
+                            }
+                            return $data;
+                        });
```

Comment:
This is broken. It previously exited the whole method early. Now it does nothing, because the returned value isn't used (see line 130).